### ISSUE TITLE
Mutect2 standard-min-confidence-threshold-for-calling

### ISSUE THREAD

**Issue by igordot** [broadinstitute/gatk#5845]
I just tried Mutect2 from GATK 4.1.1.0 and got an error:
```
A USER ERROR has occurred: standard-min-confidence-threshold-for-calling is not a recognized option
```

From the [online documentation](https://software.broadinstitute.org/gatk/documentation/tooldocs/4.1.0.0/org_broadinstitute_hellbender_tools_walkers_mutect_Mutect2.php):

> Note that the default was changed from 10.0 to 30.0 in version 4.1.0.0 to accompany the switch to use the the new quality score by default.

Thus, it was still maintained in 4.1.0.0. Based on that, I am surprised that it was removed.

There are actually a few other parameters that got dropped. For example, `normal-artifact-lod` from `FilterMutectCalls`. Are these changes explained somewhere?

**Comment by ldgauthier** [comment]
That argument is only relevant to HaplotypeCaller and GenotypeGVCFs. It
should never have been available for Mutect2.

**Comment by davidbenjamin** [comment]
@igordot It used to exist because `AssemblyBasedCallerArgumentCollection` used to extend `StandardCallerArgumentCollection`, causing `Mutect2` to have a bunch of `HaplotypeCaller` arguments that it didn't use.  This was fixed in PR #5758.

`FilterMutectCalls` also lost a few arguments as part of a huge change to the entire filtering model in PR #5688.  I'm working on a blog post about this but for now the Mutect2 docs at https://github.com/broadinstitute/gatk/blob/master/docs/mutect/mutect.pdf are up-to-date and more user-friendly than they used to be.

**Comment by davidbenjamin** [comment]
If you get any more of these errors, it's either an argument that never had any effect or something that you 4.1.1 got rid of.  In the latter case, you don't need to replace it with anything.  In 4.1.1 `FilterMutectCalls` automatically learns a lot of parameters.
